examples/deepspeed/huggingface/Dockerfile

```diff
@@ -0,0 +1,11 @@
+FROM determinedai/model-hub-transformers:0.18.4-dev0 as model_hub
+
+RUN mkdir /wheels && pip wheel -w /wheels model_hub==0.18.4
+
+FROM determinedai/environments:cuda-11.3-pytorch-1.10-tf-2.8-deepspeed-0.7.0-gpu-0.19.4 as runtime
+
+ADD requirements.txt /tmp/requirements.txt
+RUN pip install -r /tmp/requirements.txt
+
+COPY --from=model_hub /wheels/model_hub-0.18.4-py3-none-any.whl /tmp/model_hub_wheels/model_hub-0.18.4-py3-none-any.whl
+RUN pip install --find-links=/tmp/model_hub_wheel model-hub
```

Comment:
@salanki - The `model_hub` source code is within this repo, `coreweave_determined/model_hub` so I don't think we could use submodules. Determined also doesn't publish the model_hub package anywhere. It looks like they build it from source everytime they create a new version of their `determinedai/model-hub-transformers` docker image.

So to get around this I just build and take the `model_hub` wheel from the `model-hub-transformers` image into the determined deepspeed image. I also included instructions in the readme about how to build model_hub from source for your local environment.